### ISSUE TITLE
Now not working was going great

### ISSUE THREAD

**Issue by ColinTitahi** [lllyasviel/ControlNet#33]
Hi Love this looks fanatic, knew there must be a way of combining open pose with diffusion.
Had it working great on a previous commit.
Rebuilt environment and a fresh pull and re downloaded models
Now if fails after 
Loaded model config from [./models/cldm_v15.yaml]
Killed

any help?

**Comment by brycedrennan** [comment]
`killed` generally means you ran out of memory

**Comment by ColinTitahi** [comment]
Thanks but my my V Ram is barley touched (almost nothing used of 12GB) but normal ram is getting quite high but does not seem to max out and it was running fine.

**Comment by brycedrennan** [comment]
yeah killed would be for normal ram getting maxed. Possible to not see in an activity monitor because it quickly kills the process.

**Comment by ColinTitahi** [comment]
Thanks - hmmm 16gb not enough?

**Comment by brycedrennan** [comment]
no idea. could be you have too many other things open. could be a bug in the code here.

**Comment by G-force78** [comment]
RAM maxxing out here too, on 12gb free collab notebooks.

**Comment by lllyasviel** [comment]
we are looking into this https://github.com/lllyasviel/ControlNet/pull/25

**Comment by joemamafue** [comment]
> no idea. could be you have too many other things open. could be a bug in the code here.



> RAM maxxing out here too, on 12gb free collab notebooks.

Forgive my ignorance, but is there a link i missed somewhere? Did you make your own Colab notebook? Could you share it and/or explain how?

**Comment by G-force78** [comment]
> > no idea. could be you have too many other things open. could be a bug in the code here.
> 
> > RAM maxxing out here too, on 12gb free collab notebooks.
> 
> Forgive my ignorance, but is there a link i missed somewhere? Did you make your own Colab notebook? Could you share it and/or explain how?
> 
> Edit: fwiw, I'm personally not concerned with training yet, just trying to get image generation working.

No worries here you go, depth map maxes out RAM the scribble map one is fine
 https://colab.research.google.com/drive/1VRrDqT6xeETfMsfqYuCGhwdxcC2kLd2P?usp=sharing

**Comment by FurkanGozukara** [comment]
you can try automatic1111 extension too

It works amazingly well and supports xformers as well now since automatic1111 supports

Also not anymore you need a merged ckpt files

It works with any custom model as well

16.) Automatic1111 Web UI
[Sketches into Epic Art with 1 Click: A Guide to Stable Diffusion ControlNet in Automatic1111 Web UI](https://www.youtube.com/watch?v=vhqqmkTBMlU) 
[![image](https://user-images.githubusercontent.com/19240467/218806127-c84d1ff8-d5bb-41b0-bdef-6922568792b9.png)
](https://www.youtube.com/watch?v=vhqqmkTBMlU)

**Comment by fkite-cs** [comment]
Hi, i faced the same problem and it is because the model is not releasing the memory after inference
i had to use:
```
torch.cuda.empty_cache()
gc.collect()
```
before the return statement in the process function

**Comment by faiza333** [comment]
@fkite-cs Can you please mention which file should i find the process function in.

**Comment by fkite-cs** [comment]
@faiza333 you can find that function in any gradio file of the repository i.e [gradio_scribble2image.py](https://github.com/lllyasviel/ControlNet/blob/d3284fcd0972c510635a4f5abe2eeb71dc0de524/gradio_scribble2image.py#L23)

**Comment by geroldmeisinger** [comment]
all duplicate concerning "RAM and out of memory exceptions (OOM)":
https://github.com/lllyasviel/ControlNet/issues/21
https://github.com/lllyasviel/ControlNet/issues/33
https://github.com/lllyasviel/ControlNet/issues/191
https://github.com/lllyasviel/ControlNet/issues/236
https://github.com/lllyasviel/ControlNet/issues/241
https://github.com/lllyasviel/ControlNet/issues/247
https://github.com/lllyasviel/ControlNet/issues/294
https://github.com/lllyasviel/ControlNet/issues/301
